elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1661,9 +1665,22 @@ protected void addInverseRelation(String relationName, Object relationValue) {
                                                                                  boolean skipNew) {
         Set<PersistentResource<T>> filteredSet = new LinkedHashSet<>();
         for (PersistentResource<T> resource : resources) {
+            Set<Triple<Class, Class, String>> expressionResultShortCircuit
+                    = resource.getRequestScope().getExpressionResultShortCircuit();
+            PermissionExecutor permissionExecutor = resource.getRequestScope().getPermissionExecutor();
+
             try {
                 if (!(skipNew && resource.getRequestScope().getNewResources().contains(resource))) {
-                    resource.checkFieldAwarePermissions(permission);
+                    if (!expressionResultShortCircuit
+                            .contains(Triple.of(permission, resource.getResourceClass(), null))) {
+                        permissionExecutor.checkUserPermissions(resource.getResourceClass(), permission);
+                        if (expressionResultShortCircuit
```

Comment:
Instead of checking whether the `expressionResultShortCircuit` contains the item (which is relying on a side effect of `checkUserPermissions`, why not just check the return value of checkUserPermissions?